src/api/CompileTools.ts

```diff
@@ -516,6 +516,12 @@ export namespace CompileTools {
         if (curlib) ileSetup.currentLibrary = curlib;
       }
 
+      // Remove any duplicates fromr the library list
+      ileSetup.libraryList = ileSetup.libraryList
+        .filter((item, pos, self) => {
+          return self.indexOf(item) == pos;
+        })
```

Comment:
For the sake of reusing the Tools at hand :)
```suggestion
      ileSetup.libraryList = ileSetup.libraryList.filter(Tools.distinct);
```

Reply:
Oh, I did not even know we had that. Fab!!